connectors/byoc.py

```diff
@@ -580,39 +498,44 @@ def next_sync(self):
         if self.sync_now:
             logger.debug("sync_now is true, syncing!")
             return 0
-        if not self.scheduling["enabled"]:
+        if not self.scheduling.get("enabled", False):
             logger.debug("scheduler is disabled")
             return SYNC_DISABLED
-        return next_run(self.scheduling["interval"])
+        return next_run(self.scheduling.get("interval"))
+
+    async def reset_sync_now_flag(self):
+        await self.index.update(doc_id=self.id, doc={"sync_now": False})
```

Comment:
Note from discussion: either update _source too, or somehow indicate that you need to reload the object so that it's up-to-date.